src/overlays/actors/ovl_Obj_Etcetera/z_obj_etcetera.c

```diff
@@ -26,35 +32,280 @@ const ActorInit Obj_Etcetera_InitVars = {
+    if (objectIndex >= 0) {
+        this->objIndex = objectIndex;
+    }
+    pos.x = this->dyna.actor.world.pos.x;
+    pos.y = this->dyna.actor.world.pos.y + 10.0f;
+    pos.z = this->dyna.actor.world.pos.z;
+    func_800C411C(&globalCtx->colCtx, &this->dyna.actor.floorPoly, &floorBgId, &this->dyna.actor, &pos);
+    this->dyna.actor.floorBgId = floorBgId;
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    this->actionFunc = ObjEtcetera_Setup;
+    Actor_SetScale(&this->dyna.actor, 0.01f);
+    this->dyna.actor.scale.y = 0.02f;
+    this->burrowFlag = 0;
+}
+
+void ObjEtcetera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjEtcetera* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void ObjEtcetera_DoNormalOscillation(ObjEtcetera* this, GlobalContext* globalCtx) {
+    if (this->oscillationTimer > 0) {
+        Actor_SetScale(&this->dyna.actor,
+                       (oscillationTable[globalCtx->gameplayFrames % 18] * (0.0001f * this->oscillationTimer)) + 0.01f);
+        this->dyna.actor.scale.y = 0.02f;
+        this->oscillationTimer--;
+    dummy_label:; // POSSIBLE FAKE MATCH
+        return;
+    }
+    Actor_SetScale(&this->dyna.actor, 0.01f);
+    this->dyna.actor.scale.y = 0.02f;
+}
+
+void ObjEtcetera_StartSmallFlutterAnimation(ObjEtcetera* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_040117A8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_040117A8.common), 2,
+                         0.0f);
+    this->dyna.actor.draw = ObjEtcetera_DrawAnimated;
+    this->actionFunc = ObjEtcetera_PlaySmallFlutterAnimation;
+}
+
+void ObjEtcetera_Idle(ObjEtcetera* this, GlobalContext* globalCtx) {
+    s16 minOscillationTimer;
+    Player* player = PLAYER;
+
+    if ((player->stateFlags3 & 0x200) && (this->dyna.actor.xzDistToPlayer < 20.0f)) {
+        // Link is launching himself out of the Deku Flower
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_0400EB7C, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0400EB7C.common), 2,
+                             0.0f);
+        this->dyna.actor.draw = ObjEtcetera_DrawAnimated;
+        this->actionFunc = ObjEtcetera_DoIntenseOscillation;
+        Actor_SetScale(&this->dyna.actor, 0.01f);
+        this->dyna.actor.scale.y = 0.02f;
+        this->intenseOscillationScale = 0.003f;
+        this->oscillationTimer = 30;
+        this->burrowFlag &= ~1;
+    } else if ((player->stateFlags3 & 0x2000) && (this->dyna.actor.xzDistToPlayer < 30.0f) &&
+               (this->dyna.actor.yDistToPlayer > 0.0f)) {
+        // Link is hovering above the Deku Flower
+        minOscillationTimer = 10 - (s32)(this->dyna.actor.yDistToPlayer * 0.05f);
+        if (this->oscillationTimer < minOscillationTimer) {
+            this->oscillationTimer = minOscillationTimer;
+        }
```

Comment:
```suggestion
        this->oscillationTimer = CLAMP_MIN(this->oscillationTimer, minOscillationTimer);
```
Would this still match?

Reply:
Nope, this doesn't match.